File: CLIESC_TICKETERA_SOAP_JAVA_GR01/src/renderer/src/mappers/SoccerGame.mapper.ts
import { SoccerGame } from '@renderer/entities/SoccerGame'
import { Mapper, SoccerGameResponse } from '@renderer/types'

export class SoccerGameMapper implements Mapper<SoccerGameResponse, SoccerGame> {
  toEntity(dto: SoccerGameResponse) {
    return new SoccerGame(
      dto.codigo,
      dto.equipoLocal,
      dto.equipoVisita,
      new Date(dto.fecha.replace(/\[.*\]$/, '')),
      dto.lugar
    )
  }
}

File: CLICON_TICKETERA_SOAP_JAVA_GR01/src/main/java/ec/edu/monster/model/mapper/SoccerGameMapper.java
package ec.edu.monster.model.mapper;

import ec.edu.monster.model.dto.SoccerGameResponse;
import ec.edu.monster.model.entity.SoccerGame;

public class SoccerGameMapper {
    public static SoccerGame toEntity(SoccerGameResponse dto) {
        if (dto == null)
            return null;

        return new SoccerGame(
                dto.codigo(),
                dto.equipoLocal(),
                dto.equipoVisita(),
                dto.fecha(), // ZonedDateTime mapeado automáticamente por Gson
                dto.lugar());
    }
}